content/posts/09-nibbler-flatpak.md

```diff
@@ -0,0 +1,230 @@
+---
+title: "Flatpak-ing Nibbler, a Leela Chess Zero Interface and some Neural Network Engines"
+date: 2020-10-30T17:39:23-03:00
+draft: false
+---
+
+> Article by editor Leo <thinkabit.ukim@gmail.com>  
+
+> The Machine Learning terminology here is simplified and will have a loss
+> quality, but not pertinent to the point.
+
+Nibbler is a Graphical User Interface for analysis of chess games using mainly
+the [Leela Chess Zero][1] chess engine (computer that plays chess and provides
+an evaluation of positions), it is written specifically for using Leela.
+
+It deals with Leela's output which is incompatible with traditional chess engines
+and the Chess GUIs that support the latter, that specific support allow it to
+make full use of Leela's output, like filtering the moves that Leela suggested
+by the certainty it has that the value given to the move is correct, and sorting
+the moves by the amount of moves left until a conclusion (win, draw or loss) to
+the game.
+
+Unfortunately for us it is written in Electron which makes it very hard to package
+for [musl][2] systems like [Alpine Linux][3] which I use and am a developer,
+fortunately there is [flatpak][4] which is a godsend for packaging heavyweight apps,
+so lets go package it.
+
+<!--more-->
+
+## Writing the boilerplate
+
+Lets start by writing the boilerplate we need to get started on any Flatpak app.
+
+```yaml
+app-id: org.fohristiwhirl.nibbler
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+branch: stable
+sdk: org.freedesktop.Sdk
+command: /app/nibbler/nibbler
+
+finish-args:
+  - --share=ipc # Needed for X11
+  - --socket=x11 # Electron application, needs Ozone, switch to fallback-x11 once chrome gets ozone
+  - --device=dri # need access to the GPU
+  - --filesystem=xdg-data # Access to $XDG_DATA_HOME, letting the user load networks
```

Comment:
networks are files that are stored in $XDG_DATA_HOME/lc0 and XDG_DATA_HOME/lc0/networks

Reply:
> letting the user load networks

Huh, why does it need access to ~/.local/share for that?